### PR TITLE
whitelist library tracks pages

### LINKED ISSUE #26: Support "Library > Tracks" scrobbles page
image

I think it's nice to support user library tracks scrobbles page of URL like `https://www.last.fm/user/{USER}/library/music/{ARTIST}/_/{TRACK}?date_preset=ALL`.

### PR DESCRIPTION
**Pull request by CennoxX**
Match path to library tracks pages in validatePage(). The paths include "username/library" and between album and track is an underscore in the form "album/_/track".
Add library tracks page to alert.
This fixes #26

### CODE REVIEW

File: unscrobbler.js [hunk 1]
```diff
@@ -177,8 +177,8 @@ function validatePage() {
   const url = new URL(document.URL);
 
   if (!username) alert('You are not logged in.\nPlease log in to use Last.FM Unscrobbler.');
-  else if (url.pathname.endsWith(username) || url.pathname.endsWith(`${username}/library`)) unscrobbler();
-  else alert('Last.FM Unscrobbler works only on:\n· Your profile page\n· Your library scrobbles page');
+  else if (url.pathname.endsWith(username) || url.pathname.endsWith(`${username}/library`) || (url.pathname.includes(`${username}/library`) && url.pathname.split('/').slice(-2)[0] == "_")) unscrobbler();
```
Comment (guytepper) [inline unscrobbler.js:180]: Nice!! 
I think it'd be good to space out the condition for readability, and also to add a comment to clear out what's the condition is for :) 

```suggestion
  else if (
    url.pathname.endsWith(username) ||
    url.pathname.endsWith(`${username}/library`) ||
    // Covers the library track page
    // The path includes "username/library" and between the album and the track there is an underscore ("album/_/track").
    (url.pathname.includes(`${username}/library`) && url.pathname.split('/').slice(-2)[0] == '_')
  )
```

File: unscrobbler.js [hunk 2]
```diff
@@ -177,8 +177,8 @@ function validatePage() {
...
+  else alert('Last.FM Unscrobbler works only on:\n· Your profile page\n· Your library scrobbles page\n· Your library tracks page');
```
Comment (guytepper) [inline unscrobbler.js:181]: The issue with "library tracks page" is that it's not the tracks page itself (not possible to use it there) 
What do you think about using the following term instead?

```suggestion
  else alert('Last.FM Unscrobbler works only on:\n· Your profile page\n· Your library scrobbles page\n· An individual track scrobble history');
```

**Review by guytepper — COMMENTED** [review]
Awesome! Thanks a lot for this!
Just some small adjustments before merging :)

I also noticed an issue with the extension styles on the track page (the checkbox appears on the right side instead of the left, and the delete selected scrobbles doesn't show in bold red - but let's tackle it in a different PR
image

File: unscrobbler.js (hunk 2 above)
Comment (CennoxX) [reply unscrobbler.js:187]: You're right, this should make it more clear.

**Comment by guytepper** [comment]
Thank you 🙏
I've opened an issue on the wrong styles, let me know if you'd like to take it too :)

**Pull request merged**